### PR TITLE
Homework3: Alexandr Anisko

### PR DESCRIPTION
**Pull request by Talion90**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework3/task1.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+""" FizzBuzz
+
+Напишите программу, которая печатает цифры от 1 до 100, но вместо чисел,
+кратных 3 пишет Fizz, вместо чисел кратный 5
+пишет Buzz, а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz
+"""
+
+for digit in range(1, 101):
+    if not digit % 3 and not digit % 5:
```
Comment (belovmd) [inline src/homework3/task1.py:9]: 15?

File: src/homework3/task2.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+1. Используйте генератор списков чтобы получить следующий: ['ab', 'ac', 'ad', 'bb', 'bc', 'bd'].
+2. Используйте на предыдущий список slice чтобы получить следующий: ['ab', 'ad', 'bc'].
+3. Используйте генератор списков чтобы получить следующий ['1a', '2a', '3a', '4a'].
+4. Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+5. Скопируйте список и добавьте в него элемент '2a' так
+чтобы в исходном списке этого элемента не было.
+"""
+
+# 1 Create list by generator
+lst = [str_1 + str_2 for str_1 in ['a', 'b'] for str_2 in ['b', 'c', 'd']]
```
Comment (belovmd) [inline src/homework3/task2.py]: 'bcd' ?

File: src/homework3/task2.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+print(lst)
+
+# 2 Create list by slice
+lst_2 = lst[::2]
+print(lst_2)
+
+# 3 Create list by generator
+lst_3 = [str(digit) + 'a' for digit in range(1, 5)]
```
Comment (belovmd) [inline src/homework3/task2.py]: '1234' ?

File: src/homework3/task3.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+1. Создайте список ['a', 'b', 'c'] и сделайте из него кортеж.
+2. Создайте кортеж ('a', 'b', 'c'), И сделайте из него список
+3. Сделайте следующие присвоения одной строкой a = 'a', b=2, c=’python’.
+4. Создайте кортеж из одного элемента, чтобы при итерировании по этому элементы последовательно
+   выводились значения 1, 2, 3. Убедитесь что len() исходного кортежа возвращает 1.
+"""
+
+# 1. Creating list and making tuple
+
+lst = list('abcd')
```
Comment (belovmd) [inline src/homework3/task3.py]: why string conversion?

File: src/homework3/task3.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+tpl = tuple(lst)
+print(lst, tpl)
+
+# 2. Creating tuple and making list
+
+tpl = tuple('abcd')
```
Comment (belovmd) [inline src/homework3/task3.py]: why should we convert str?

File: src/homework3/task4.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+равные друг другу образуют одну пару, которую необходимо посчитать.
+Входные данные - строка из чисел, разделенная пробелами.
+Выходные данные - количество пар.
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+"""
+
+import random
+
+# Создаем список чисел для входных данных
+# lst = list(str(random.randint(10000000, 99999999)))
```
Comment (belovmd) [inline src/homework3/task4.py]: redundant comments

File: src/homework3/task4.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+counted_numbers = []
+pairs = []
+
+# Считаем количество повторяющихся элементов
+for num in lst:
+    if num not in counted_numbers:
+        counted_numbers.append(num)
+        pairs.append(lst.count(num))
+    else:
+        pass
```
Comment (belovmd) [inline src/homework3/task4.py]: redundant code

File: src/homework3/task5.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+"""Уникальные элементы в списке
+
+Дан список. Выведите те его элементы, которые встречаются в списке только один раз.
+Элементы нужно выводить в том порядке, в котором они встречаются в списке.
+"""
+
+lst = ['a', 1, 2, 3, 3, 1, 'b', 'z', '1', 'a', 0]
+for element in lst:
+    if lst.count(element) > 1:
+        pass
```
Comment (belovmd) [inline src/homework3/task5.py]: why do we need it?

Comment (belovmd) [reply src/homework3/task5.py]: can we invers condition?

File: src/homework3/task6.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+# Создаем список чисел для входных данных
+lst = [random.randint(-5, 5) for num in range(20)]
+print('Входные данные: ', lst)
+
+# Перемещаем все нули в правую часть списка за один проход
+for num in lst:
+    if not num:
+        lst.append(lst.pop(lst.index(num)))
+    else:
+        pass
```
Comment (belovmd) [inline src/homework3/task6.py]: redundant code

File: src/homework3/task6.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+задачу нужно выполнить за один проход по списку. Распечатайте полученный список.
+"""
+
+import random
+
+# Создаем список чисел для входных данных
+lst = [random.randint(-5, 5) for num in range(20)]
+print('Входные данные: ', lst)
+
+# Перемещаем все нули в правую часть списка за один проход
```
Comment (belovmd) [inline src/homework3/task6.py]: redundant comments

**Review by belovmd — CHANGES_REQUESTED** [review]

File: src/homework3/task1.py (hunk 1 above)
Comment (Talion90) [reply src/homework3/task1.py:9]: print fizzbuzz, i had test it before pushing

File: src/homework3/task2.py (hunk 2 above)
Comment (Talion90) [reply src/homework3/task2.py]: also we can write ...in 'ab' for str_2 in 'bcd'

File: src/homework3/task2.py (hunk 3 above)
Comment (Talion90) [reply src/homework3/task2.py]: yes, really easier=)

File: src/homework3/task1.py (hunk 1 above)
Comment (belovmd) [reply src/homework3/task1.py:9]: not digit % 15

**Review by belovmd — APPROVED** [review]
8,5

**Pull request closed without merging**